### PR TITLE
Bump versions of ruby, gems and dockerfile dependencies

### PR DESCRIPTION
**Pull request by manelmontilla**

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.5.9
+3.0.0
```
Comment (danfaizer) [inline .ruby-version:1]: I would suggest to use same version than the one used in the Dockerfile.
```bash
3.1.2
```

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,11 @@
 language: ruby
 rvm:
-  - 2.5.9
+  - 3.1.0
```
Comment (danfaizer) [inline .travis.yml:3]: Same as above ☝️ 
I've verified that [3.1.2 version is available](https://rubies.travis-ci.org/) in Travis:

File: Dockerfile [hunk 3]
```diff
@@ -1,17 +1,17 @@
 # Copyright 2019 Adevinta
 
 ARG rails_env=production
-FROM ruby:2.5.9-alpine AS pre-builder
+FROM ruby:3.1-alpine3.15 AS pre-builder
```
Comment (danfaizer) [inline Dockerfile:4]: I would suggest to fix patch version as well:
```
3.1.2-alpine3.15
```
So local and CI matches the version.

File: Dockerfile [hunk 4]
```diff
@@ -21,16 +21,16 @@ RUN bundle install -j4 --retry 3 \
     && find /gems/ -name "*.c" -delete \
     && find /gems/ -name "*.o" -delete
 
-FROM ruby:2.5.9-alpine
+FROM ruby:3.1-alpine3.15
```
Comment (danfaizer) [inline Dockerfile:24]: Same as above ☝️

File: .ruby-version (hunk 1 above)
Comment (manelmontilla) [reply .ruby-version]: Yes you are completely right thank you.

File: .travis.yml (hunk 2 above)
Comment (manelmontilla) [reply .travis.yml]: Perfect thanks, I'm modifying it also.

File: Dockerfile (hunk 3 above)
Comment (manelmontilla) [reply Dockerfile]: yes I agree it's better.

File: Dockerfile (hunk 4 above)
Comment (manelmontilla) [reply Dockerfile]: 👍

**Review by danfaizer — APPROVED** [review]
Thanks for the update @manelmontilla 
👍

**Pull request merged**